migrations/20250227_viz_move/migration.py

```diff
@@ -0,0 +1,249 @@
+import os
+import shutil
+import re
+import redis
+import logging
+
+
+def setup_logging():
+    """Configure logging for the migration script."""
+    logging.basicConfig(
+        level=logging.INFO,
+        format="%(asctime)s - %(levelname)s - %(message)s",
+        datefmt="%Y-%m-%d %H:%M:%S",
+    )
+    return logging.getLogger("migration")
+
+
+def create_backup(base_folder, logger):
+    """Create backups of all output folders before migration."""
+    logger.info("Creating backups of output folders...")
+
+    shutil.copytree(base_folder, base_folder + "_backup")
+    
+    logger.info("Created backup of poppunk_output")
+    
+    
+def get_output_folders(base_folder):
+    # Get all valid output folders
+    all_output_folders = [
+        os.path.join(base_folder, item)
+        for item in os.listdir(base_folder)
+        if os.path.isdir(os.path.join(base_folder, item))
+    ]
+
+    # Filter for folders with network folder (i.e old output folders to update)
+    output_folders = [
+        folder
+        for folder in all_output_folders
+        if os.path.exists(os.path.join(folder, "network"))
+    ]
+    return output_folders
+
+def update_redis_keys(logger):
+    """Update Redis job keys to match new structure."""
+    logger.info("Updating Redis keys...")
+    r = redis.Redis(host="beebop-redis")
+
+    # Rename the Redis key from "beebop:hash:job:microreact" to "beebop:hash:job:visualise"
+    if r.exists("beebop:hash:job:microreact"):
+        microreact_data = r.hgetall("beebop:hash:job:microreact")
+        for field, value in microreact_data.items():
+            r.hset("beebop:hash:job:visualise", field, value)
```

Comment:
This isn't actually renaming the key  is it? It's leaving the microreact key and values as they are, but just copying them into the visualise key. Which is fair enough since the visualise key may already exist by the time you run the migration, and I'm not sure you'd want to do a RENAME in that case. (Plus it wouldn't be rollback-able.) So maybe just change the comment and log message as they're a little misleading..?